src/pages/dashboard/withDashboardController.tsx

```diff
@@ -48,46 +57,15 @@ export const loadingInit = {
 };
 
 let RECENT_ORDER_PAGE_INCREASE = 1;
-let ORDER_TIMELINE_PAGE_INCREASE = 1;
 
 export const withDashboardController = <P,>(Component: ComponentType<P>): ComponentType<P> => {
   return (props: P) => {
-    const { dailyOrder, dailyRevenue, dailyCustomer, orderTimeline, recentOrder, filter } = useObservable(
-      dashboardStore.model,
-    );
-    const [loading, setLoading] = useState<LoadingTypes>(loadingInit);
-
-    const fetchChartData = async (params: DailyParams): Promise<void> => {
-      setLoading((prev) => ({ ...prev, dailyCustomer: true, dailyOrder: true, dailyRevenue: true }));
-      try {
-        Promise.all([
-          await DashboardService.instance.fetchDailyRevenue(params),
-          await DashboardService.instance.fetchDailyOrder(params),
-          await DashboardService.instance.fetchDailyCustomer(params),
-        ]);
-      } catch (e) {
-        console.error(e);
-      } finally {
-        setLoading((prev) => ({ ...prev, dailyCustomer: false, dailyOrder: false, dailyRevenue: false }));
-      }
-    };
-
-    const fetchTableData = async (): Promise<void> => {
-      setLoading((prev) => ({ ...prev, orderTimeline: true, recentOrder: true }));
-      try {
-        await DashboardService.instance.fetchBothRecentOrderAndOrderTimeline({
-          page: DEFAULT_PARAMS.PAGE,
-          pageSize: DEFAULT_PARAMS.PAGE_SIZE,
-          sorters: DEFAULT_PARAMS.SORTERS,
-        });
-      } catch (e) {
-        console.error(e);
-      } finally {
-        RECENT_ORDER_PAGE_INCREASE++;
-        ORDER_TIMELINE_PAGE_INCREASE++;
-        setLoading((prev) => ({ ...prev, orderTimeline: false, recentOrder: false }));
-      }
-    };
+    const { orderTimeline, filter, recentOrderPage } = useObservable(dashboardStore.model);
+    const { data: dailyRevenue, isLoading: dailyRevenueLoading } = useFetchDailyRevenue(filter);
```

Comment:
Please ensure that the filter is not null. 
Add enabled for all this hook.

ex:
const { data: dailyRevenue, isLoading: dailyRevenueLoading } = useFetchDailyRevenue(filter, {
  **enabled: !!filter**,
});